qiskit_nature/problems/sampling/protein_folding/bead_distances/distance_map_builder.py

```diff
@@ -0,0 +1,292 @@
+        side_chain = peptide.get_side_chain_hot_vector()
+        for lower_bead_ind in range(1, main_chain_len):  # upper_bead_ind>lower_bead_ind
+            for upper_bead_ind in range(lower_bead_ind + 1, main_chain_len + 1):
+                lower_main_bead, lower_side_bead = self._get_main_and_side_beads(
+                    lower_bead_ind, peptide
+                )
+                upper_main_bead, upper_side_bead = self._get_main_and_side_beads(
+                    upper_bead_ind, peptide
+                )
+
+                upper_indic_funs = self._get_indicator_funs(peptide, side_chain, upper_bead_ind)
+                lower_indic_funs = self._get_indicator_funs(peptide, side_chain, lower_bead_ind)
+
+                self._calc_dists_main_side_all_axes(
+                    upper_indic_funs,
+                    lower_bead_ind,
+                    lower_main_bead,
+                    peptide,
+                    upper_bead_ind,
+                    upper_side_bead,
+                )
+
+                self._calc_dists_side_main_all_axes(
+                    lower_bead_ind,
+                    lower_indic_funs,
+                    lower_side_bead,
+                    peptide,
+                    upper_bead_ind,
+                    upper_main_bead,
+                )
+
+                self._calc_dists_side_side_all_axes(
+                    upper_indic_funs,
+                    lower_bead_ind,
+                    lower_indic_funs,
+                    lower_side_bead,
+                    peptide,
+                    upper_bead_ind,
+                    upper_side_bead,
+                )
+
+    @staticmethod
+    def _get_main_and_side_beads(bead_ind: int, peptide: Peptide) -> Tuple[MainBead, SideBead]:
+        main_bead = peptide.get_main_chain[bead_ind - 1]
+        if main_bead.side_chain:
+            side_bead = main_bead.side_chain[0]
+        else:
+            side_bead = None
+        return main_bead, side_bead
+
+    @staticmethod
+    def _get_indicator_funs(
+        peptide: Peptide, side_chain: List[bool], bead_ind: int
+    ) -> Union[
+        Tuple[None, None, None, None], Tuple[OperatorBase, OperatorBase, OperatorBase, OperatorBase]
+    ]:
+        if side_chain[bead_ind - 1]:
+            indic_0, indic_1, indic_2, indic_3 = (
+                peptide.get_main_chain[bead_ind - 1].side_chain[0].indicator_functions
+            )
+        else:
+            indic_0, indic_1, indic_2, indic_3 = None, None, None, None
+        return indic_0, indic_1, indic_2, indic_3
+
+    def _calc_dists_side_side_all_axes(
+        self,
+        upper_indic_funs: Tuple[OperatorBase, OperatorBase, OperatorBase, OperatorBase],
+        lower_bead_ind: int,
+        lower_indic_funs: Tuple[OperatorBase, OperatorBase, OperatorBase, OperatorBase],
+        lower_side_bead: BaseBead,
+        peptide: Peptide,
+        upper_bead_ind: int,
+        upper_side_bead: BaseBead,
```

Comment:
This also seems like a rather awkward order...
Same in the next method.. I will refrain from more comments like this but please check other methods, too.